yt_dlp/postprocessor/ffmpeg.py

```diff
@@ -914,3 +974,129 @@ def run(self, info):
+                    f'Skipping {self.pp_key()} since the video was already downloaded. '
+                    'Use --force-remove-chapters to remove the chapters anyway')
+                return [], info
+
+        concat_opts = self._make_concat_opts(ranges_to_cut, self._get_video_duration(info))
+
+        def remove_chapters(file, is_sub):
+            return file, self.remove_chapters(file, ranges_to_cut, concat_opts, self._force_keyframes and not is_sub)
+
+        in_out_files = [remove_chapters(info['filepath'], False)]
+
+        sub_files = []
+        for _, sub in (info.get('requested_subtitles') or {}).items():
+            sub_file = sub.get('filepath')
+            # The file might have been removed by --embed-subs
+            if not sub_file or not os.file.exists(sub_file):
+                continue
+            ext = sub['ext']
+            if ext not in self.SUPPORTED_SUBS:
+                self.report_warning(f'Cannot remove chapters from external {ext} subtitles; "{sub_file}" is now out of sync')
+                continue
+            # TODO: create __real_download for subs
+            sub_files.append(sub_file)
+
+        in_out_files.extend(remove_chapters(in_file, True) for in_file in sub_files)
+
+        # Renaming should only happen after all files are processed
+        files_to_remove = []
+        for in_file, out_file in in_out_files:
+            uncut_file = prepend_extension(in_file, 'uncut')
+            if os.path.exists(uncut_file):
+                os.remove(uncut_file)
+            os.rename(encodeFilename(in_file), encodeFilename(uncut_file))
+            os.rename(encodeFilename(out_file), encodeFilename(in_file))
+            files_to_remove.append(uncut_file)
+
+        return files_to_remove, info
+
+    @staticmethod
+    def _remove_chapters_from_infodict(*chapters_list, regexes):
+        combined_chapters = ((n, c) for n, chapters in enumerate(chapters_list) for c in chapters)
+
+        removed_sections, keep_chapters = [], [[] for _ in chapters_list]
+        for i, (n, c) in enumerate(sorted(combined_chapters, key=lambda x: x[1]['start_time'])):
+            assert c['end_time'] >= c['start_time']
+            if c.get('title') and any(regex.search(c['title']) for regex in regexes):
+                if removed_sections and removed_sections[-1][1] >= c['start_time']:
+                    removed_sections[-1][1] = max(c['end_time'], removed_sections[-1][1])
+                else:
+                    removed_sections.append([c['start_time'], c['end_time']])
+            else:
+                keep_chapters[n].append(c)
+
+        def get_removed_time_before(time):
+            return sum(min(time, end) - start for start, end in removed_sections if start < time)
+
+        for chapters in keep_chapters:
+            for c in chapters:
+                c['start_time'] -= get_removed_time_before(c['start_time'])
+                c['end_time'] -= get_removed_time_before(c['end_time'])
```

Comment:
Two loops (inner loop in sum): quadratic time.

Reply:
I can think of some ways to improve this using `itertools`, but I am not really convinced that looking at speed here is worth it. Even with 1000 cuts and double that many chapters (worst case), the loop runs under ~60ms on my laptop. Realistically, we'll only ever be dealing with < 100 chapters and much fewer cuts